### PR TITLE
asciidoctor: add asciidoctor-bibliography

### PR DESCRIPTION
**Pull request by Twey**
(change via bundix)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by evils** [comment]
this should probably be merged to staging (i'm switching my #87646 to that now)

i haven't tried the result, but since i think closure size was a concern for https://github.com/NixOS/rfcs/pull/64, this takes `asciidoctor`'s `nix path-info -S` from `337247856` to `374326416`

**Comment by evils** [comment]
apparently i spoke too soon and staging is recommended for PRs that rebuild >1000 packages

**Comment by Twey** [comment]
So should this be onto `staging`? :thinking:

**Comment by evils** [comment]
> So should this be onto `staging`?

i've been told otherwise
i was thinking it may be useful to not rebuild the dependents twice
but my PR is separate enough from this that they're not likely to be merged around the same time anyway

**Comment by Twey** [comment]
Great :)

**Comment by SuperSandro2000** [comment]
This does not need to go trough staging.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 87882` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages marked as broken and skipped:</summary>
  <ul>
    <li>almanah</li>
    <li>atom</li>
    <li>atom-beta</li>
    <li>btrbk</li>
    <li>gnome-recipes</li>
    <li>gnome3.nautilus-python</li>
    <li>mate.mate-utils</li>
    <li>udiskie</li>
  </ul>
</details>
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>asciidoctor</li>
    <li>bat-extras.batgrep</li>
    <li>esh</li>
    <li>newsboat</li>
    <li>ripgrep</li>
    <li>ripgrep-all</li>
    <li>styx</li>
    <li>weechat</li>
    <li>weechat-unwrapped</li>
  </ul>
</details>

**Comment by FlorianFranzen** [comment]
This should probably now go to the version with extensions.

File: pkgs/tools/typesetting/asciidoctor/Gemfile.lock [hunk 1]
```diff
@@ -93,4 +116,4 @@ DEPENDENCIES
   rouge
 
 BUNDLED WITH
-   2.1.4
+   1.17.3
```
Comment (SuperSandro2000) [inline pkgs/tools/typesetting/asciidoctor/Gemfile.lock:119]:
```suggestion
   2.1.4
```

**Comment by SuperSandro2000** [comment]
Closing in favor of the linked PR

**Pull request closed without merging**